blueprints/data-solutions/data-platform-minimal/02-composer.tf

```diff
@@ -0,0 +1,122 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# tfdoc:file:description Cloud Composer resources.
+
+locals {
+  env_variables = {
+    BQ_LOCATION        = var.location
+    CURATED_BQ_DATASET = module.cur-bq-0.dataset_id
+    CURATED_GCS        = module.cur-cs-0.url
+    CURATED_PRJ        = module.cur-project.project_id
+    DP_KMS_KEY         = var.service_encryption_keys.compute
+    DP_REGION          = var.region
+    GCP_REGION         = var.region
+    LAND_PRJ           = module.land-project.project_id
+    LAND_GCS           = module.land-cs-0.name
+    PHS_CLUSTER_NAME   = module.processing-dp-historyserver.name
+    PROCESSING_GCS     = module.processing-cs-0.name
+    PROCESSING_PRJ     = module.processing-project.project_id
+    PROCESSING_SA_DP   = module.processing-sa-dp-0.email
+    PROCESSING_SUBNET  = local.processing_subnet
+    PROCESSING_VPC     = local.processing_vpc
+  }
+}
+
+module "processing-sa-cmp-0" {
+  source       = "../../../modules/iam-service-account"
+  project_id   = module.processing-project.project_id
+  prefix       = var.prefix
+  name         = "prc-cmp-0"
+  display_name = "Data platform Composer service account"
+  iam = {
+    "roles/iam.serviceAccountTokenCreator" = [local.groups_iam.data-engineers]
+    "roles/iam.serviceAccountUser"         = [module.processing-sa-cmp-0.iam_email]
+  }
+}
+
+resource "google_composer_environment" "processing-cmp-0" {
+  count   = var.composer_config.disable_deployment == true ? 0 : 1
+  project = module.processing-project.project_id
+  name    = "${var.prefix}-prc-cmp-0"
+  region  = var.region
+  config {
+    software_config {
+      airflow_config_overrides = try(var.composer_config.software_config.airflow_config_overrides, null)
+      pypi_packages            = try(var.composer_config.software_config.pypi_packages, null)
+      env_variables = merge(
+        try(var.composer_config.software_config.env_variables, null), local.env_variables
+      )
+      image_version = var.composer_config.software_config.image_version
+    }
+    dynamic "workloads_config" {
```

Comment:
You can remove the dynamic block since `var.composer_config.workloads_config` is always non-null